File: agm-frontend/src/app/features/admin/services/profile.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from '@angular/core';
import { ApiResponse } from "@core/models/api-response.model";
import { Profile } from "@core/models/profile.model";
import { environment } from "@env/environment";
import { ProfileEditRequest } from "@features/assembly/models/profile-edit-request";
import { Observable } from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class ProfileService {
  private apiUrl = environment.apiUrl;

  constructor(private http: HttpClient) { }

  getProfiles(): Observable<ApiResponse<Profile[]>> {
    return this.http.get<ApiResponse<Profile[]>>(`${this.apiUrl}/profiles`);
  }

  addProfile(profileAddRequest: { name: string; permissionIds: number[] }): Observable<ApiResponse<Profile>> {
    return this.http.post<ApiResponse<Profile>>(`${this.apiUrl}/profiles`, profileAddRequest);
  }

  updateProfile(id: number, profileEditRequest: ProfileEditRequest): Observable<ApiResponse<Profile>> {
    return this.http.patch<ApiResponse<Profile>>(`${this.apiUrl}/profiles/${id}`, profileEditRequest);
  }
}

File: agm-backend/src/main/java/ma/cimr/agmbackend/service/ProfileService.java
package ma.cimr.agmbackend.service;

import java.util.List;

import ma.cimr.agmbackend.dto.request.ProfileAddRequest;
import ma.cimr.agmbackend.dto.request.ProfileEditRequest;
import ma.cimr.agmbackend.dto.response.ProfileResponse;

public interface ProfileService {

	List<ProfileResponse> getProfiles();

	ProfileResponse getProfile(Long id);

	ProfileResponse createProfile(ProfileAddRequest request);

	ProfileResponse updateProfile(Long id, ProfileEditRequest profileEditRequest);

	void updatePermissions(Long profileId, List<Long> permissionIds);
}
